### PR TITLE
Split state in SyncJobsActor

### PR DESCRIPTION
**Pull request by EgorKulikov**
We need to split state in SyncJobActor to free up ClientActor to do other things in the meantime (very important for validators)

### Test plan

This still requires full integration test

### CODE REVIEW

**Review by bowenwang1996 — APPROVED** [review]
Looks good to me on the high level. @mzhangmzz could you review the details?

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -4336,6 +4348,24 @@ pub struct BlockCatchUpResponse {
     pub results: Vec<Result<ApplyChunkResult, Error>>,
 }
 
+#[derive(Message)]
+#[rtype(result = "()")]
+pub struct StateSplitRequest {
+    pub sync_hash: CryptoHash,
+    pub shard_id: ShardId,
```
Comment (mzhangmzz) [inline chain/chain/src/chain.rs:4355]: nit: shard_id can be get from shard_uid from shard_uid.shard_id(), so it's a little redundant to have both here

**Review by mzhangmzz — APPROVED** [review]
Looks great! Thanks for putting together a PR so fast!

File: chain/client/src/test_utils.rs [hunk 2]
```diff
@@ -1486,14 +1486,20 @@ pub fn run_catchup(
...
+    let state_split_messages = Arc::new(RwLock::new(vec![]));
+    let state_split_inside_messages = state_split_messages.clone();
+    let state_split = move |msg: StateSplitRequest| {
+        state_split_inside_messages.write().unwrap().push(msg);
+    };
+    let rt = client.runtime_adapter.clone();
     while !client.chain.store().iterate_state_sync_infos().is_empty() {
-        let call = client.run_catchup(highest_height_peers, &f, &block_catch_up)?;
-        for msg in messages.write().unwrap().drain(..) {
+        let call = client.run_catchup(highest_height_peers, &f, &block_catch_up, &state_split)?;
```
Comment (mzhangmzz) [inline chain/client/src/test_utils.rs:1501]: I didn't appreciate why you put the logic for sending requests to a closure until now. This makes testing way easier. Great design!

**Comment by bowenwang1996** [comment]
@chefsale @frol @khorolets please review :)

**Review by chefsale — APPROVED** [review]

**Review by khorolets — APPROVED** [review]

**Pull request merged**